plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt

```diff
@@ -12,29 +12,37 @@ import com.mapbox.android.core.location.LocationEngineResult
 import com.mapbox.android.core.permissions.PermissionsManager
 import com.mapbox.common.Logger
 import com.mapbox.geojson.Point
+import com.mapbox.maps.plugin.PuckBearingSource
+import com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings
 import java.lang.ref.WeakReference
 import java.util.concurrent.CopyOnWriteArrayList
 
 /**
  * Default Location Provider implementation, it can be overwritten by users.
  */
-internal class LocationProviderImpl(context: Context) :
-  LocationProvider, LocationEngineCallback<LocationEngineResult> {
+internal class LocationProviderImpl(context: Context, var settings: LocationComponentSettings) :
+  LocationProvider,
+  LocationEngineCallback<LocationEngineResult>,
+  LocationCompassEngine.CompassListener {
   private val contextWeekRef: WeakReference<Context> = WeakReference(context)
   private val locationEngine = LocationEngineProvider.getBestLocationEngine(context)
-
+  private val locationCompassEngine = LocationCompassEngine(context)
   private val locationEngineRequest =
     LocationEngineRequest.Builder(LocationComponentConstants.DEFAULT_INTERVAL_MILLIS)
       .setFastestInterval(LocationComponentConstants.DEFAULT_FASTEST_INTERVAL_MILLIS)
       .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY)
       .build()
 
   private val locationConsumers = CopyOnWriteArrayList<LocationConsumer>()
-
+  private var currentPuckBearingSource: PuckBearingSource = PuckBearingSource.COURSE
```

Comment:
What if user will specify `PuckBearingSource.HEADING` let's say in XML - do I get it correctly that we will still apply `COURSE` when creating this class and then update it to `HEADING` when  `updateSettings` will be called? is it expected behaviour? won't it lead to some glitches?

Reply:
We invoke `updateCompass()` in the `init` block, so it should work well. I also double confirmed it with test example.